quinn/src/endpoint.rs

```diff
@@ -116,6 +120,17 @@ where
         Ok(())
     }
 
+    /// Replace the server configuration, affecting new incoming connections only
+    ///
+    /// Useful for e.g. refreshing TLS certificates without disrupting existing connections.
```

Comment:
You should already be able to do this with the current config, since the cert is not necessarily static but a lookup callback. Same applies for other functions for which "providers" can be hooked in (like the CidGenerator).

I guess it won't hurt to swap the complete config instead of just adding more callbacks. But the question is whether swapping this config dynamically might break anything that doesn't expect the values to change over time.

Reply:
> You should already be able to do this with the current config

Good point. Setters are a lot more convenient than setting up a sidechannel, though, and it might be useful to adjust some of the other options on the fly non-disruptively as well.

> But the question is whether swapping this config dynamically might break anything

I've audited this and I'm satisfied that there aren't any issues with the current set of config options. We actually don't read the server config in very many places; the overwhelming bulk is in `Endpoint::handle_first_packet`. At worst, swapping out `token_key` might cause spurious failure for in-flight handshakes, but that's trivially recoverable and far less disruptive than restarting the endpoint outright. It's also nice for rotating `token_key` to be enabled in its own right.

What could break things would be changing the config for an existing connection, but existing connections hold onto the `Arc` of the config they were created with, so that's not a risk.